lualib/lua_scanners/common.lua

```diff
@@ -381,7 +381,8 @@ local function gen_extension(fname)
 
   local ext = {}
   for n = 1, 2 do
-      ext[n] = #filename_parts > n and string.lower(filename_parts[#filename_parts + 1 - n]) or nil
+      ext[n] = #filename_parts > n 
+        and string.lower(string.gsub(filename_parts[#filename_parts + 1 - n],'[%c%s%p]','')) or nil
```

Comment:
Could you please add a descriptive comment of what you are changing here. For me, it is not very intuitive atm.